### DISCUSSION TITLE
Filter in converter Jar

### DISCUSSION THREAD

**Discussion by gaurav-aggarwal0301** [async-profiler/async-profiler#643] [Q&A] [answered]
Do we have any mechanism in converter Jar to filter few of the profiles not required during Flamegraph generation?

**Comment by apangin** [comment, marked as answer]
Currently you can do that in a few steps:
1. Convert `.jfr` to `.collapsed` format using `jfr2flame` converter with `--collapsed` option.
2. Filter `.collapsed` output as you wish: it's just a simple text format, you can easily process it with any handy tool like `grep`.
3. Finally, convert `.collaped` to a flame graph using `FlameGraph` script from the same `converter.jar`.

It should be fairly easy to add a filter option to `jfr2flame` to do everything in just one step. I'll consider that for one of the future releases.

**Reply by gaurav-aggarwal0301** [reply]
Thanks for explaining the way to achieve the job however what basis we can filter the internal threads or unwanted threads in collapsed file. If I compare the flame graph generated via JMC and converter jar both looks totally different. Converter jar has lot of extra threads and events which JMC doesn't show.

**Reply by apangin** [reply]
Added `--include`/`--exclude` options to `FlameGraph` and `jfr2flame` converters.
These options specify regex pattern to filter stack traces.
